src/client/datascience/kernel-launcher/kernelLauncher.ts

```diff
@@ -104,23 +102,9 @@ export class KernelLauncher implements IKernelLauncher {
         timeout: number,
         resource: Resource,
         workingDirectory: string,
-        ui: IDisplayOptions,
         cancelToken: CancellationToken
     ): Promise<IKernelProcess> {
         const promise = (async () => {
-            // If this is a python interpreter, make sure it has ipykernel
```

Comment:
Are we planning on just assuming things are installed and attempt to start the kernel?
I.e. are we planning on figuring out if ipykernel is installed or not only based on kernel failure?

I think that could be a little dangerous. I'm very concerned that analyzing errors to determine whether ipykernel is installed or not  might not be bullet proof.

Also, in the case of non-raw we call `ensureKernelIsUsable` which in turn calls `installMissingDependencies`.
Hence we should have a consistent approach for both.

Reply:
Jupyter is different in that we can't let the kernel startup fail as we can't really parse the output of the failure. But for raw, this speeds up kernel startup by like 3 seconds. Failure should be the odd ball case.

Perhaps the failure analyzer could use the kernelDependencyService to detect instead of looking at the message.

Reply:
> erhaps the failure analyzer could use the kernelDependencyService to detect instead of looking at the message.

Yes, that's what the error handler did. Having the analyzer do that would also work.

Reply:
This raises a concern I had
* Based on the output we display errors in the cell output using displayErrorInCell
* Later we display the error message in the error handler to display a messagebox

Basically we have two places where we decide what to do with the errors, can we centralize this (that's what the ErrorHandler did).
Would be great if we can merge `displayErrorInCell` with the `handleError` in the errorhandler.
This way, what we do with the errors is handled in a single location.
Whether we display errors in the cell output and/or display a message box is all in one spot - ie. error handling is centralized

Reply:
That doesn't work because there are two ways to handle errors. 
- Interactive window
- Regular notebook

The caller of the error handler has to know where to put the error.

Reply:
Yeah it does seem weird to show the message info in the error handler though. Feels more like the error handler should just return the failure info and let the calling code decide what to do with it.

Reply:
I think we should discuss options for 'where' to display the errors. There are two cases at least (interactive and notebook).

I'm not sure the errorHandler is the right spot to show any UI